File: scripts/cleanSchema.py
import re

file_path = "prisma/schema.prisma"
with open(file_path, "r", encoding="utf-8") as f:
    content = f.read()

# Remove the models
models_to_remove = [
    "StrategicStage",
    "StrategicStageStep",
    "GovernanceStage",
    "GovernanceStageStep",
    "FinanceStage",
    "FinanceStageStep"
]

for model in models_to_remove:
    # Pattern to match: model ModelName { ... }
    # Using regex with DOTALL to match the whole block
    pattern = r"model\s+" + model + r"\s+\{.*?\}"
    content = re.sub(pattern, "", content, flags=re.DOTALL)

# Remove lines in Charity model
content = re.sub(r"^\s*strategicStage\s+Int.*?$", "", content, flags=re.MULTILINE)
content = re.sub(r"^\s*strategicStages\s+StrategicStage\[\].*?$", "", content, flags=re.MULTILINE)
content = re.sub(r"^\s*governanceStages\s+GovernanceStage\[\].*?$", "", content, flags=re.MULTILINE)
content = re.sub(r"^\s*financeStages\s+FinanceStage\[\].*?$", "", content, flags=re.MULTILINE)

# Clean up multiple blank lines
content = re.sub(r"\n{3,}", "\n\n", content)

with open(file_path, "w", encoding="utf-8") as f:
    f.write(content)

print("Schema updated successfully.")

File: scripts/cleanSchema.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, '../prisma/schema.prisma');
let content = fs.readFileSync(filePath, 'utf-8');

const modelsToRemove = [
  'StrategicStage',
  'StrategicStageStep',
  'GovernanceStage',
  'GovernanceStageStep',
  'FinanceStage',
  'FinanceStageStep'
];

for (const model of modelsToRemove) {
  const pattern = new RegExp(`model\\s+${model}\\s+\\{[^\\}]+\\}`, 'g');
  content = content.replace(pattern, '');
}

content = content.replace(/^\s*strategicStage\s+Int.*?$/gm, '');
content = content.replace(/^\s*strategicStages\s+StrategicStage\[\].*?$/gm, '');
content = content.replace(/^\s*governanceStages\s+GovernanceStage\[\].*?$/gm, '');
content = content.replace(/^\s*financeStages\s+FinanceStage\[\].*?$/gm, '');

content = content.replace(/\n{3,}/g, '\n\n');

fs.writeFileSync(filePath, content, 'utf-8');
console.log('Schema updated successfully');
